src/Core/Helpers/PathHelper.cs

```diff
@@ -149,5 +151,38 @@ private static string ToFilePathString(Uri uri)
             string escapedPath = uri.OriginalString.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
             return Uri.UnescapeDataString(escapedPath);
         }
+
+
+        /// <summary>
+        /// Gets Temp Directory under SLVS folder under %Temp%
+        /// </summary>
+        /// <param name="perVSInstance">Determines if the path will be per VS Instance</param>
+        /// <param name="paths">Creates sub folder structure under SLVS folder</param>
+        public static string GetTempDirForTask(bool perVSInstance, params string[] paths)
+        {
+            var SLVSTempFolder = Path.Combine(Path.GetTempPath(), "SLVS");
+
+            var taskPaths = new List<string> { SLVSTempFolder };
+
+            if (paths != null)
+            {
+                foreach (var path in paths)
+                {
+                    if (!string.IsNullOrWhiteSpace(path))
```

Comment:
We don't need to code defensively here; we're the only people who use this, so it we're passing in bad arguments then it's a problem we should fix.

It would be enough to assert e.g.
```
Debug.Assert(!folders.Any(x => string.IsNullOrWhiteSpace(x)), "Supplied folder name  should not be null/empty");
```

Reply:
If we don't do that "Path.Combine" will throw an error. We can add an assert here, but I'd rather not to throw an exception.

Reply:
If we were using values supplied by the end user or from a dynamic source that could fail then we would need to be defensive i.e. if there are potential runtime issues to guard against.

Here, the paths are constructed from static data. If a null is passed in then it's because there is a bug in the our calling code. We'd rather know about it sooner than later, so it's appropriate for an exception to be thrown.

We should code defensively against runtime issues. We shouldn't code defensively against our own code being badly written.